### PR TITLE
Add localizedName and localizedDescription to block definition

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Add localizedName and localizedDescription to BlockDefinition. This includes a DB evolution to populate the english translation to be the existing name/description and also updates the program service to set the localized values when the english versions are updated on the admin side.

This does not surface translations to user, or add the UI for admins to update the translations.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

#### Database evolutions

Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#writing-database-evolutions)

- [x] Assigned two reviewers
- [x] Guarded against already existing resources using `IF NOT EXISTS` and `IF EXISTS`
- [x] Downs created to undo changes in Ups

### Issue(s) this completes

Progress towards #7606

### CODE REVIEW

**Comment by greg-hyde** [comment]
* **#7804** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7804?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/conf/evolutions/default/69.sql [hunk 1]
```diff
@@ -0,0 +1,52 @@
+-- These assume a default locale of en_US, which is true for all deployments as of 2023-09-05.
```
Comment (avaleske) [inline server/conf/evolutions/default/69.sql:1]: Also true as of june 2024

**Review by avaleske — APPROVED** [review]

File: server/conf/evolutions/default/69.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        ELSE block_definition -- Keep the block_definition as is
+    END
+  )
+  FROM jsonb_array_elements(p.block_definitions) AS block_definition
+)
+WHERE block_definitions IS NOT NULL AND jsonb_array_length(block_definitions) <> 0;
+
+
+# --- !Downs
+UPDATE ONLY programs
```
Comment (gwendolyngoetz) [inline server/conf/evolutions/default/69.sql:40]: What does the ONLY do?

**Review by gwendolyngoetz — APPROVED** [review]
I'm not particularly familiar with some of the json methods, but it seems fine to me.

File: server/conf/evolutions/default/69.sql (hunk 1 above)
Comment (greg-hyde) [reply server/conf/evolutions/default/69.sql]: updated :)

File: server/conf/evolutions/default/69.sql (hunk 2 above)
Comment (greg-hyde) [reply server/conf/evolutions/default/69.sql]: i had borrowed it from 57.sql, but it's not needed, so I removed it

**Pull request merged**